scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java

```diff
@@ -2039,6 +2044,34 @@ public void newJobSubmitted(final InternalJob job) {
         });
     }
 
+    private void replaceSystemVariables(Serializable job) {
```

Comment:
I understand the reason why you give a Serializable job parameter, but the code would be more elegant by creating two private methods, one which takes InternalJob and the other one which takes JobData

Reply:
I was thinking about this too but there would be 2 methods with the exact same code, the only difference would be the method parameter InternalJob/JobData

Reply:
if you think that it would look better, I can do the change